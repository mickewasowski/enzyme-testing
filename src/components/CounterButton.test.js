import React from "react";
import { shallow } from "enzyme";
import CounterButton from './CounterButton';

it('expect to render CounterButton component', () => {
  const mockColor = 'cyan';
  expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
});

it('correctly updates the counter', () => {
  const mockColor = 'cyan';
  const wrapper = shallow(<CounterButton color={mockColor} />);

  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 1 });
  expect(wrapper.props().color).toEqual('cyan');
});

