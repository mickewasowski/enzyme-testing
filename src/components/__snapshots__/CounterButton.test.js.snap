// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CounterButton component 1`] = `
<button
  color="cyan"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
